x/cellarfees/keeper/keeper.go

```diff
@@ -58,10 +65,46 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 	k.paramSpace.SetParamSet(ctx, &params)
 }
 
-/////////////////////
-// Module Accounts //
-/////////////////////
+////////////////////////////////
+// Last highest reward supply //
+////////////////////////////////
 
-func (k Keeper) GetFeesAccount(ctx sdk.Context) authtypes.ModuleAccountI {
-	return k.accountKeeper.GetModuleAccount(ctx, types.ModuleName)
+func (k Keeper) GetLastRewardSupplyPeak(ctx sdk.Context) sdk.Int {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(types.GetLastRewardSupplyPeakKey())
+	if b == nil {
+		panic("Last highest reward supply should not have been nil")
+	}
+	var amount big.Int
+	return sdk.NewIntFromBigInt((&amount).SetBytes(b))
+}
+
+func (k Keeper) SetLastRewardSupplyPeak(ctx sdk.Context, amount sdk.Int) {
+	store := ctx.KVStore(k.storeKey)
+	b := amount.BigInt().Bytes()
+	store.Set(types.GetLastRewardSupplyPeakKey(), b)
+}
+
+//////////////////////////
+// Fee accrual counters //
+//////////////////////////
+
+func (k Keeper) GetFeeAccrualCounters(ctx sdk.Context) (counters types.FeeAccrualCounters) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(types.GetFeeAccrualCountersKey())
+	if b == nil {
+		panic("Fee accrual counters is nil, it should have been set by InitGenesis")
+	}
+	if len(b) == 0 {
+		return types.DefaultFeeAccrualCounters()
```

Comment:
If the value must have been set in genesis, why do we have a case that returns a default?

Reply:
Sure, but I guess my question is, when would `len(b) == 0` ever resolve to true? That's saying there is nothing stored for fee accrual counters. Shouldn't there always be a counter from genesis?

Reply:
As strange as it seems, genesis is setting it to an empty struct but its length is zero when you retrieve it. You can show this by running the unit tests with and without that check commented out.

Reply:
So to confirm, setting this to an empty struct returns a byte array of length zero rather than a strictly nil value? In go AFAIK, `len(nil)` also returns zero, and in some modules I have seen one check covering both cases. If we can differentiate from nil and an empty struct having been set, having the `panic` on nil is probably fine.

Reply:
My unit tests fail without the 0 length check:

```go
=== RUN   TestKeeperTestSuite/TestKeeperGettingSettingFeeAccrualCounters
    keeper_test.go:103:
        	Error Trace:	/Users/atro/source/work/sommelier/x/cellarfees/keeper/keeper_test.go:103
        	Error:      	Not equal:
        	            	expected: types.FeeAccrualCounters{Counters:[]types.FeeAccrualCounter{}}
        	            	actual  : types.FeeAccrualCounters{Counters:[]types.FeeAccrualCounter(nil)}

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,3 @@
        	            	 (types.FeeAccrualCounters) {
        	            	- Counters: ([]types.FeeAccrualCounter) {
        	            	- }
        	            	+ Counters: ([]types.FeeAccrualCounter) <nil>
        	            	 }
```

Reply:
Yeah, I guess I'm just trying to ascertain what the conditions are inside the keeper function that might result in two different return values. What conditions will cause this function to actually panic? Since `Get` returns a byte slice, is it that never having set a value at this key ever will result in it returning nil, versus having set it with an empty value `Get` will return a zero-length nil slice?

Reply:
If a value is never set, it panics at line 96 (commenting out setting value in genesis shows this), setting it with an empty value will return a struct with a nil slice